### ISSUE TITLE
No mention of advanced token generation scenario that ties to specific user id

### ISSUE THREAD

**Issue by drub0y** [MicrosoftDocs/bot-docs#808]
When generating a token a more advanced scenario is supported where you can specify exactly what userId the token should be tied to and that scenario is not documented here. It's covered here in this blog post[1] from Sept 2018, but I would expect the official documentation to cover this as well.

This is a very important security scenario because specifying the userId via the WebChat initialization logic is extremely insecure.

[1] https://blog.botframework.com/2018/09/25/enhanced-direct-line-authentication-features/

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9fd0bdf8-7cb8-1f26-65a1-3876ea84351f
* Version Independent ID: bda6ce3a-f006-baaf-db50-82bd79afa6ab
* Content: [Authentication - Bot Service](https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-authentication?view=azure-bot-service-4.0#get-a-direct-line-secret)
* Content Source: [articles/rest-api/bot-framework-rest-direct-line-3-0-authentication.md](https://github.com/MicrosoftDocs/bot-docs/blob/live/articles/rest-api/bot-framework-rest-direct-line-3-0-authentication.md)
* Service: **bot-service**
* Product: **bot-framework**
* GitHub Login: @RobStand
* Microsoft Alias: **kamrani**

**Comment by GraniteStateHacker** [comment]
Is there anything else we can add to the token other than User.Id?   I would love to add a 3rd party token, perhaps filling in the jwt's InnerToken …?

**Comment by mutanttech** [comment]
If some one does not controls the customer portals where bot can be hooked using the web chat. In that case how it will be feasible to generate the Directline Token, without atleast passing the secret once from the JS. Please include this case as well.

**Comment by JonathanFingold** [comment]
This is now partially addressed with the addition of a discussion about [Web Chat and Direct Line considerations](https://docs.microsoft.com/azure/bot-service/bot-builder-authentication?view=azure-bot-service-4.0#web-chat-and-direct-line-considerations). A dedicated article about this scenario is planned.

**Comment by compulim** [comment]
Agree. This article should be updated with contents from [this blog](https://blog.botframework.com/2018/09/25/enhanced-direct-line-authentication-features/) to make it complete.

Anyone who is interested, please feel free to submit PR to this article. Doc team will help proofreading and correct grammars.

And I believe your GitHub profile will appear as a contributor on this doc, which is a great credit.

**Comment by mmiele** [comment]
Work in progress.
See https://github.com/MicrosoftDocs/bot-docs-pr/pull/1680.

**Comment by mmiele** [comment]
@drub0y Please review the fixes in this staged document; https://review.docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-direct-line-3-0-authentication?view=azure-bot-service-3.0&branch=pr-en-us-1680#security-considerations
